### PR TITLE
Add swager to our project #62

### PR DESCRIPTION
**Pull request by NovitskaMariia**
1) run: pip install -r requirements.txt
2) run: python manage.py runserver
3) go to:   http://127.0.0.1:8000/yasg/swagger/   and scroll down
4) test 2 sections: USER REGISTER, LOG_IN, LOG-OUT and RESET PASSWORD.
    to test it, you should click on the endpoint and click on the 'Try it out' botton.

image

### CODE REVIEW

File: forum/yasg.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+#       title="Your Project API",
+#       default_version='v1',
+#       description="API Description",
+#    ),
+#    public=True,
+# )
+
+# urlpatterns = [
+#     path('swagger/', schema_view.with_ui('swagger', config='UA_1155_alpha/swagger-config.yaml'), name='schema-swagger-ui'),
+# ]
```
Comment (superolegatron) [inline forum/yasg.py:42]: Maybe delete these comments if we do not need this code in future

File: users/views.py [hunk 2]
```diff
@@ -17,6 +17,88 @@
...
+    }
+    Example error response (wrong password):
+    HTTP 401 Unauthorized
+    {
+        "message": "Wrong password"
+    }
+    """
+
+    @swagger_auto_schema(
+        security=[],
```
Comment (superolegatron) [inline users/views.py:53]: Maybe we can move everything from @swagger_auto_schema(...) into another file it will improve readability of the code

**Review by superolegatron — COMMENTED** [review]
Delete unnecessry comments and try to investigate whether can we move code from swagger decorators into another file or not

File: forum/urls.py [hunk 3]
```diff
@@ -26,6 +27,9 @@
 )
 from startups.views import StartupViewSet
 
+# from .yasg import urlpatterns as yasg_urls
```
Comment (VovaD199) [inline forum/urls.py:30]: Maybe delete these comments if we do not need this code in future

File: forum/yasg.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
+from django.urls import path
+from rest_framework import permissions
+from drf_yasg.views import get_schema_view
+from drf_yasg import openapi
+
+...
```
Comment (VovaD199) [inline forum/yasg.py:6]: Maybe delete these comments if we do not need this code in future

File: users/views.py (hunk 2 above)
Comment (NovitskaMariia) [reply users/views.py:53]: Done :)

File: users/swagger_auto_schema_settings.py [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+    },
+    example={
+        "email": "testforum97@gmail.com",
+        "password": "Password123!",
+    },
+)
+
+loginAPIView_responses = {
+    200: openapi.Response(
+        description="User created successfully",
```
Comment (YuliaShap) [inline users/swagger_auto_schema_settings.py:22]: Maybe we should write here "User login successfully"

**Review by YuliaShap — APPROVED** [review]

**Review by superolegatron — APPROVED** [review]
Cool

**Pull request merged**